admin/services/class-indexable-term-provider.php

```diff
@@ -10,49 +10,141 @@
-
+	public function get( $object_id, $as_object = false ) {
 		if ( ! $this->is_indexable( $object_id ) ) {
 			return array();
 		}
 
-		return array(
-			'object_id'                   => (int) $object_id,
-			'object_type'                 => 'term',
-			'object_subtype'              => $term->taxonomy,
-			'permalink'                   => get_term_link( $term ),
-			'canonical'                   => $this->get_meta_value( 'canonical', $term ),
-			'title'                       => $this->get_meta_value( 'title', $term ),
-			'description'                 => $this->get_meta_value( 'desc', $term ),
-			'breadcrumb_title'            => $this->get_meta_value( 'bctitle', $term ),
-			'og_title'                    => $this->get_meta_value( 'opengraph-title', $term ),
-			'og_description'              => $this->get_meta_value( 'opengraph-description', $term ),
-			'og_image'                    => $this->get_meta_value( 'opengraph-image', $term ),
-			'twitter_title'               => $this->get_meta_value( 'twitter-title', $term ),
-			'twitter_description'         => $this->get_meta_value( 'twitter-description', $term ),
-			'twitter_image'               => $this->get_meta_value( 'twitter-image', $term ),
-			'is_robots_noindex'           => $this->get_robots_noindex_value( $this->get_meta_value( 'noindex', $term ) ),
-			'is_robots_nofollow'          => null,
-			'is_robots_noarchive'         => null,
-			'is_robots_noimageindex'      => null,
-			'is_robots_nosnippet'         => null,
-			'primary_focus_keyword'       => $this->get_meta_value( 'focuskw', $term ),
-			'primary_focus_keyword_score' => (int) $this->get_meta_value( 'linkdex', $term ),
-			'readability_score'           => (int) $this->get_meta_value( 'content_score', $term ),
-			'is_cornerstone'              => false,
-			'link_count'                  => null,
-			'incoming_link_count'         => null,
-			'created_at'                  => null,
-			'updated_at'                  => null,
-		);
+		$indexable = WPSEO_Term_Indexable::from_object( $object_id );
+
+		if ( $as_object === true ) {
+			return $indexable;
+		}
+
+		return $indexable->to_array();
+	}
+
+	/**
+	 * Handles the patching of values for an existing indexable.
+	 *
+	 * @param int   $object_id   The ID of the object.
+	 * @param array $requestdata The request data to store.
+	 *
+	 * @return void
+	 *
+	 * @throws WPSEO_Invalid_Indexable_Exception The indexable exception.
+	 * @throws Exception 						 Exception that is thrown if patching the object has failed.
+	 */
+	public function patch( $object_id, $requestdata ) {
+		$indexable = $this->get( $object_id, true );
+
+		if ( $indexable === array() ) {
+			throw WPSEO_Invalid_Indexable_Exception::non_existing_indexable( $object_id );
+		}
+
+		$new_indexable    = $indexable->update( $requestdata );
+		$stored_indexable = $this->store_indexable( $new_indexable );
+
+		if ( $stored_indexable === true ) {
+			return;
+		}
+
+		throw new Exception( 'Patch failed' );
```

Comment:
This should preferably return a more accurate message, more importantly it should be translatable.